internal/services/containers/kubernetes_cluster_resource_test.go

```diff
@@ -513,6 +533,87 @@ resource "azurerm_kubernetes_cluster" "test" {
   `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, controlPlaneVersion, tag)
 }
 
+func (KubernetesClusterResource) azureKeyVaultKms(data acceptance.TestData, controlPlaneVersion string, enabled bool) string {
+	kmsBlock := ""
+	if enabled {
+		kmsBlock = `
+  key_management_service {
+    key_vault_key_id = azurerm_key_vault_key.test.id
+  }`
+	}
```

Comment:
Reused existing code style.

Earlier commit used the dependencies `"bytes"` and `"text/template"` which is simpler for adding and removing hcl blocks. Since no other test uses Golang templating I removed it.

Are you opinionated for one or the other solution, @stephybun ?

Reply:
Since we don't use Golang templating for test configs I would have suggested it be removed anyway, so current approach is good 👍